File: smart-label/frontend/src/api/material.ts
import request from "@/utils/request";

export type Album = "oral" | "skin";

export interface MaterialPhoto { rel_path: string; filename: string; size_bytes: number }
export interface MaterialDogFolder { name: string; photos: MaterialPhoto[] }
export interface MaterialDateFolder { folder: string; date: string | null; ok: boolean; dogs: MaterialDogFolder[] }

/** 素材库相册（日期 > 狗 > 照片）：oral=口腔验证，skin=颈圈算法验证/皮肤瘙痒/视频问诊 */
export const listMaterialPhotos = (album: Album) => request.get<never, { root: string; folders: MaterialDateFolder[] }>(`/material/${album}/photos`);
export const getMaterialPhotoToken = (album: Album, path: string) => request.post<never, { token: string }>(`/material/${album}/photos/token`, { path });
export const materialPhotoUrl = (album: Album, path: string, token: string) =>
  `/api/v1/material/${album}/photos/stream?path=${encodeURIComponent(path)}&token=${token}`;

File: smart-label/backend/app/api/v1/material.py
"""
素材库相册浏览（只读）：album=oral 口腔照片 / skin 皮肤瘙痒问诊照片，两者在 NAS 上都是
{日期-ok}/{狗}/*.jpg 的结构，共用 tooth_service 里的扫描/路径签名逻辑。牙齿识别页用
/tooth/photos（带检测结果），皮肤评估页用这里的 /material/skin/photos（纯看图）。
"""

from fastapi import APIRouter, Depends, HTTPException, Request, status

from app.core.deps import require_role
from app.models.user import UserRole
from app.schemas.envelope import ok
from app.services import tooth_service as svc
from app.services.range_stream import stream_file

router = APIRouter(prefix="/material", tags=["material"], dependencies=[Depends(require_role(UserRole.admin, UserRole.super_admin))])
stream_router = APIRouter(prefix="/material", tags=["material"])

_CONTENT_TYPES = {".jpg": "image/jpeg", ".jpeg": "image/jpeg", ".png": "image/png", ".webp": "image/webp", ".bmp": "image/bmp"}


@router.get("/{album}/photos")
async def list_photos(album: str):
    try:
        return ok({"root": svc.album_root(album), "folders": svc.list_photos(album)})
    except svc.ToothError as e:
        raise HTTPException(status.HTTP_404_NOT_FOUND, str(e)) from e


@router.post("/{album}/photos/token")
async def photo_token(album: str, body: dict):
    rel_path = body.get("path") or ""
    try:
        svc.resolve_photo(rel_path, album)
    except svc.ToothError as e:
        raise HTTPException(status.HTTP_404_NOT_FOUND, str(e)) from e
    return ok({"token": svc.issue_photo_token(rel_path, album)})


@stream_router.get("/{album}/photos/stream")
async def photo_stream(album: str, path: str, token: str, request: Request):
    if not svc.verify_photo_token(path, token, album):
        raise HTTPException(status.HTTP_403_FORBIDDEN, "token 无效或已过期")
    try:
        full = svc.resolve_photo(path, album)
    except svc.ToothError as e:
        raise HTTPException(status.HTTP_404_NOT_FOUND, str(e)) from e
    ext = full[full.rfind("."):].lower()
    return await stream_file(request, full, _CONTENT_TYPES.get(ext, "application/octet-stream"))
